instrumentation/jetty/jetty-8.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/v8_0/JettyHandlerTest.java

```diff
@@ -0,0 +1,172 @@
+    options.setHasResponseSpan(endpoint -> endpoint == REDIRECT || endpoint == ERROR);
+    options.setExpectedException(new IllegalStateException(EXCEPTION.getBody()));
+    super.configure(options);
+  }
+
+  @Override
+  protected SpanDataAssert assertResponseSpan(
+      SpanDataAssert span, String method, ServerEndpoint endpoint) {
+    if (endpoint == REDIRECT) {
+      span.satisfies(spanData -> assertThat(spanData.getName()).endsWith(".sendRedirect"));
+    } else if (endpoint == ERROR) {
+      span.satisfies(spanData -> assertThat(spanData.getName()).endsWith(".sendError"));
+    }
+    span.hasKind(SpanKind.INTERNAL).hasAttributesSatisfying(Attributes::isEmpty);
+    return span;
+  }
+
+  private static void handleRequest(Request request, HttpServletResponse response) {
+    ServerEndpoint endpoint = ServerEndpoint.forPath(request.getRequestURI());
+    controller(
+        endpoint,
+        () -> {
+          try {
+            return extracted(request, response, endpoint);
+          } catch (IOException e) {
+            throw new RuntimeException(e);
+          }
+        });
+  }
+
+  private static HttpServletResponse extracted(
+      Request request, HttpServletResponse response, ServerEndpoint endpoint) throws IOException {
+    response.setContentType("text/plain");
+    switch (endpoint) {
+      case SUCCESS:
+        response.setStatus(endpoint.getStatus());
+        response.getWriter().print(endpoint.getBody());
+        break;
+      case QUERY_PARAM:
+        response.setStatus(endpoint.getStatus());
+        response.getWriter().print(request.getQueryString());
+        break;
+      case REDIRECT:
+        response.sendRedirect(endpoint.getBody());
+        break;
+      case ERROR:
+        response.sendError(endpoint.getStatus(), endpoint.getBody());
+        break;
+      case CAPTURE_HEADERS:
+        response.setHeader("X-Test-Response", request.getHeader("X-Test-Request"));
+        response.setStatus(endpoint.getStatus());
+        response.getWriter().print(endpoint.getBody());
+        break;
+      case EXCEPTION:
+        throw new IllegalStateException(endpoint.getBody());
+      case INDEXED_CHILD:
+        INDEXED_CHILD.collectSpanAttributes(name -> request.getParameter(name));
+        response.setStatus(endpoint.getStatus());
+        response.getWriter().print(endpoint.getBody());
+        break;
+      default:
+        response.setStatus(NOT_FOUND.getStatus());
+        response.getWriter().print(NOT_FOUND.getBody());
+        break;
+    }
+    return response;
+  }
+
+  private static class TestHandler extends AbstractHandler {
+    @Override
+    public void handle(
+        String target,
+        Request baseRequest,
+        HttpServletRequest request,
+        HttpServletResponse response)
+        throws IOException, ServletException {
+      if (baseRequest.getDispatcherType() != DispatcherType.ERROR) {
```

Comment:
Can you add https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/1649a90e29cb5fad7924a9896a21be5ef525c4a6/instrumentation/jetty/jetty-8.0/javaagent/src/test/groovy/JettyHandlerTest.groovy#L133-L135 back? We've had instrumentations before that decorated the `HttpServletResponse`, even though Jetty internally depends on the fact that it's actually supposed to be `Response`

Reply:
It was fixed -- I think we should leave that one cast/assertion as a regression test though.